core/src/main/scala/kafka/log/LogCleanerManager.scala

```diff
@@ -210,6 +210,16 @@ private[log] class LogCleanerManager(val logDirs: Array[File], val logs: Pool[To
     }
   }
 
+  def truncateCheckpoint(dataDir: File, topicAndPartition: TopicAndPartition, offset: Long): Unit = {
```

Comment:
In Kafka, we use procedure syntax for methods that return `Unit` (even though I actually prefer the approach you used here).